File: app/src/main/java/me/zzy/redot/MyApplication.java
package me.zzy.redot;

import android.app.Application;
import android.content.Context;
import android.content.SharedPreferences;

import androidx.preference.PreferenceManager;

import me.zzy.redot.util.SharedPreferencesUtils;

/**
 * @author ZZY
 * 12/20/20.
 */
public class MyApplication extends Application {
    private static SharedPreferences defaultSharedPreferences;
    private static SharedPreferences historyPreferences;

    @Override
    public void onCreate() {
        super.onCreate();
        // 初始化本地缓存对象
        initSharedPreferences();

        // 初始化本地数据库对象
//        AppDatabase.getInstance(this);

    }

    private void initSharedPreferences() {
        SharedPreferencesUtils.getInstance();
        //默认的SharedPreferences
        defaultSharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);
        //自定义的SharedPreferences
        historyPreferences = getSharedPreferences(getString(R.string.preference_file_key), Context.MODE_PRIVATE);
//        SharedPreferencesUtils.getInstance
        //初始化数据
//        SharedPreferencesUtils.getInstance().put(defaultSharedPreferences, "asd", "sd");
//        SharedPreferencesUtils.getInstance().put(historyPreferences, "asd", "sd");
    }
}

File: app/src/main/java/me/zzy/redot/MyApplication.kt
package me.zzy.redot

import android.app.Application
import me.zzy.redot.util.SharedPreferencesUtils

/**
 * @author ZZY
 * 12/20/20.
 */
class MyApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        // 初始化本地缓存对象
        initSharedPreferences()

        // 初始化本地数据库对象
        AppDatabase.getDatabase(this)
    }

    private fun initSharedPreferences() {
        SharedPreferencesUtils.Companion.init(this)
        SharedPreferencesUtils.Companion.setSharedPreferences(R.string.preference_file_key)
                .put("is_scroll_forever", true)
                .put("times", Int.MAX_VALUE)
                .put("text_size", 16f)
                .put("text_color", 0xffffff)
                .put("background_color", 0x000000)
                .put("text", "test")
                .put("speed", 10)
                .put("isHorizontal", true)
                .put("clickEnable", true)
                .apply()
        //        SharedPreferencesUtils
//                .setSharedPreferences(getPackageName() + "_preferences")
//                .finish();
    }
}